lib/puppet/provider/package/pip.rb

```diff
@@ -11,7 +13,10 @@
   This provider supports the `install_options` attribute, which allows command-line flags to be passed to pip.
   These options should be specified as an array where each element is either a string or a hash."
 
-  has_feature :installable, :uninstallable, :upgradeable, :versionable, :install_options, :targetable
+  has_feature :installable, :uninstallable, :upgradeable, :versionable, :version_ranges, :install_options, :targetable
+
+  PipVersion   = Puppet::Util::Package::Version::Pip
+  VersionRange = Puppet::Util::Package::Version::Range
```

Comment:
this will conflict with apt.rb VersionRagne
```
puppet/lib/puppet/provider/package/apt.rb:9: warning: already initialized constant VersionRange
puppet/lib/puppet/provider/package/pip.rb:19: warning: previous definition of VersionRange was here
```